### ISSUE TITLE
Link labels cannot be empty

### ISSUE THREAD

**Issue by tomByrer** [remarkjs/remark#450]
### empty Link Labels should not be parsed

via CommonMark, but should be logical regardless:

> A link label begins with a left bracket ([) and ends with the first right
bracket (]) that is not backslash-escaped. Between these brackets there must
be **at least one non-whitespace** character.

### Steps to reproduce

```js
const fs = require('fs')
const remark = require("remark")
const inspect = require('unist-util-inspect')

const tree = remark().parse(`[ ] brackets with white space inside`)
console.log(inspect(tree))
```
```sh
$ node view-ast.js
root[1] (1:1-1:37, 0-36)
└─ paragraph[2] (1:1-1:37, 0-36)
   ├─ linkReference[1] (1:1-1:4, 0-3) [identifier=" "][label=" "][referenceType="shortcut"]
   │  └─ text: " " (1:2-1:3, 1-2)
   └─ text: " brackets with white space inside" (1:4-1:37, 3-36)
```

More through test; seems GFM Task List is OK with ignoring the Linked List:
https://astexplorer.net/#/gist/51df6d1a522d5a08c3017d92549345f9/ac9ddaca577f5115e7e220c74acbe786ab3a4a11

### Expected behaviour

```sh
$ node view-ast.js
root[1] (1:1-1:37, 0-36)
└─ paragraph[2] (1:1-1:37, 0-36)
   └─ text: "[ ] brackets with white space inside" (1:1-1:37, 3-36)
```

cheers, & thanks for all your hard work!

**Comment by wooorm** [comment]
Thanks for the issue! This is indeed a bug, but we are working on an alternative to the internals of remark here: #439

**Comment by tomByrer** [comment]
Thanks for your reply.

Yes, I saw micromark, but seems it won't be ready for another 6+ months as a 'new internals', & this is bug might be a blocker for me.

Either way, another test case to add to the pile.

**Comment by wooorm** [comment]
Sorry for the wait! I just wanted to share that there’s now a PR that solves this issue: https://github.com/remarkjs/remark/pull/536.

**Comment by wooorm** [comment]
This is now released in [`remark@13.0.0`](https://github.com/remarkjs/remark/releases/tag/13.0.0)
